src/Microsoft.TemplateEngine.Cli/TelemetryHelper.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TemplateEngine.Abstractions;
+
+namespace Microsoft.TemplateEngine.Cli
+{
+    public static class TelemetryHelper
+    {
+        // Checks if the input parameter value is a valid choice for the parameter, and returns the canonical value, or defaultValue if there is no appropriate canonical value.
+        // If the parameter or value is null, return defaultValue.
+        // For this to return other than the defaultValue, one of these must occur:
+        //  - the input value must either exactly match one of the choices (case-sensitive)
```

Comment:
self review: to be consistent with parameter handling when matching templates, this should be case insensitive comparisons on the parameter values.